servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletService.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.servlet;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.servlet.http.HttpServlet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpResponseWriter;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * An {@link HttpService} which handles {@link HttpRequest} and forward to Servlet APIs,
+ * and write {@link HttpResponse} to client.
+ */
+public class DefaultServletService implements HttpService {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultServletService.class);
+    private final ServletContextImpl servletContext;
+
+    /**
+     * A class which helps a {@link DefaultServletService} have a {@link HttpServlet}.
+     */
+    public DefaultServletService(ServletContextImpl servletContext) {
+        requireNonNull(servletContext, "servletContext");
+        this.servletContext = servletContext;
+    }
+
+    /**
+     * handles {@link HttpRequest} and forward to Servlet APIs.
+     */
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        final HttpResponseWriter res = HttpResponse.streaming();
+        req.aggregate().handle((aReq, cause) -> {
+            if (cause != null) {
+                logger.warn("{} Failed to aggregate a request:", ctx, cause);
+                if (res.tryWrite(ResponseHeaders.of(HttpStatus.INTERNAL_SERVER_ERROR))) {
+                    res.close();
+                }
+                return null;
+            }
+            process(res, ServletHttpExchange.of(servletContext, ctx, aReq));
```

Comment:
Is there any reason to make `ServletHttpExchange`?
We have `ServletContextImpl` already, couldn't we just make `ServletHttpRequest` and `ServletHttpResponse` directly?

Reply:
`ServletHttpExchange` like a `HttpExchange` https://docs.oracle.com/javase/8/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/HttpExchange.html in JDK HttpServer. `ServletHttpExchange` store protocol information, channel or reader/writer API. I think we should separate initialization role (`ServletContextImpl`) and communication role (`ServletHttpExchange`) . How do you think?

Reply:
Yes, but it looks like all `ServletHttpExchange` doing is just calling `getServletContext()` which we already have.
How about introducing it later when we find that it really needs?

Reply:
Agree, I will remove it.